### PR TITLE
Submit filing package

### PR DESCRIPTION
**Pull request by TayGov**
# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/mappers/EfilingFilingPackageMapper.java [hunk 1]
```diff
@@ -30,15 +28,52 @@
...
+    @Mapping(target = "payments", source = "payments")
+    @Mapping(target = "statuses", source = "statuses")
+    @Mapping(target = "packageSeqNo", source = "index")
+    EfilingDocument toEfilingDocument(Document document, Submission submission, Integer index,
+                                      String serverName, List<EfilingDocumentMilestone> milestones,
+                                      List<EfilingDocumentPayment> payments, List<EfilingDocumentStatus> statuses);
+
+    @Mapping(target = "entUserId", source = "submission.clientId")
+    @Mapping(target = "entDtm",  expression = "java(ca.bc.gov.open.jag.efilingcommons.utils.DateUtils.getCurrentXmlDate())")
+    @Mapping(target = "milestoneDtm",  expression = "java(ca.bc.gov.open.jag.efilingcommons.utils.DateUtils.getCurrentXmlDate())")
```
Comment (alexjoybc) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/mappers/EfilingFilingPackageMapper.java:53]: Nice

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/mappers/EfilingFilingPackageMapper.java [hunk 2]
```diff
@@ -30,15 +28,52 @@
...
+    @Mapping(target = "milestoneTypeCd",  constant = "ASUB")
```
Comment (alexjoybc) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/mappers/EfilingFilingPackageMapper.java:54]: Would you mind extracting all the constants to keys? So we have quick access to all those values?

Comment (TayGov) [reply src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/mappers/EfilingFilingPackageMapper.java:54]: This one is just there till we get the way to set the value from Jaime. But we can make a it a constant

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/service/SubmissionServiceImpl.java [hunk 3]
```diff
@@ -27,6 +27,8 @@
 import java.util.UUID;
 import java.util.stream.Collectors;
 
+
+
```
Comment (akroon3r) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/service/SubmissionServiceImpl.java:31]: I thought it was 2 lines between import and class definition. weird your linter didn't catch this, or I'm dumb.

File: src/backend/libs/efiling-commons/src/main/java/ca/bc/gov/open/jag/efilingcommons/model/EfilingDocument.java [hunk 4]
```diff
@@ -298,7 +302,31 @@ public void setXmlValidYn(Boolean xmlValidYn) {
...
+        return statuses;
+    }
+
+    public void setStatuses(List<EfilingDocumentStatus> statuses) {
+        this.statuses = statuses;
+    }
+
+    public EfilingDocument(Boolean amendsAnotherDocumentYn, XMLGregorianCalendar appearanceDt, XMLGregorianCalendar appearanceTm, BigDecimal appearnceAgenId, BigDecimal bailiffId, String clientFileNameTxt, String conferenceRequestTypeCd, String documentDescriptionTxt, BigDecimal documentId, String documentSubtypeCd, String documentTypeCd, XMLGregorianCalendar entDtm, String entUserId, BigDecimal estimatedToHearDocHrsQty, BigDecimal estimatedToHearDocMinsQty, String expertWitnessCallModeCd, Boolean feeExemptYn, String filePath, String fileServer, String filingBodyCd, String largeFileTimeMilliSecs, String largeFileYn, Boolean mediationStepYn, BigDecimal packageId, BigDecimal packageSeqNo, XMLGregorianCalendar updDtm, String updUserId, String uploadStateCd, String uploadedToApplicationCd, Boolean xmlDataCompleteYn, Boolean xmlDocumentInstanceYn, Boolean xmlValidYn, List<EfilingDocumentMilestone> milestones, List<EfilingDocumentPayment> payments, List<EfilingDocumentStatus> statuses) {
```
Comment (akroon3r) [inline src/backend/libs/efiling-commons/src/main/java/ca/bc/gov/open/jag/efilingcommons/model/EfilingDocument.java:329]: I guess there's no way around this atm?

File: src/backend/libs/efiling-commons/src/main/java/ca/bc/gov/open/jag/efilingcommons/model/EfilingIdentificationDetails.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+    public String getThirdGivenNm() {
+        return thirdGivenNm;
+    }
+
+    public void setThirdGivenNm(String thirdGivenNm) {
+        this.thirdGivenNm = thirdGivenNm;
+    }
+
+    public EfilingIdentificationDetails(XMLGregorianCalendar entDtm, String entUserId, String firstGivenNm, BigDecimal identificationDetailSeqNo, String nameTypeCd, String organizationNm, BigDecimal partyId, String secondGivenNm, String surnameNm, String thirdGivenNm) {
```
Comment (akroon3r) [inline src/backend/libs/efiling-commons/src/main/java/ca/bc/gov/open/jag/efilingcommons/model/EfilingIdentificationDetails.java:98]: same

**Pull request merged**